leaphymicropython/utils/i2c_helper.py

```diff
@@ -68,6 +81,84 @@ def select_channel(i2c, multiplexer_address, channel_number):
         print("Invalid channel number. Please select a channel between 0 and 7 or 255.")
 
 
+class I2CSensorOrActuator():
+    """
+    Base class for I2C sensors and actuators.
+
+    This class provides a common interface and functionality for interacting with
+    I2C devices. It includes methods for initializing the I2C bus, finding the
+    device on the bus, and selecting a channel on an I2C multiplexer if one is used.
+    """
+    MULTIPLEXER_ADDRESS = None
+    ADDRESS = None
+
+    def __init__(self):
+        self.reinitialize = None
+        self.mugs_used = None
+        self.i2c = None
+        self.channel = None
+        self.bus_id = None
+        self.scl_gpio_pin = None
+        self.sda_gpio_pin = None
+
+    def initialize_i2c(self):
+        """
+        Initializes the I2C bus.
+
+        This method sets up the I2C bus with the specified ID, SCL pin, and SDA pin.
+        It is called during the initialization of the I2C device.
+        """
+        self.i2c = I2C(
+            id=self.bus_id, scl=Pin(self.scl_gpio_pin), sda=Pin(self.sda_gpio_pin)
+        )
+
+    def find_device(self, show_warnings=True):
+        """Finds the I2C device on the bus.
+
+        This method checks if the device is visible on the I2C bus.
+        If a multiplexer is used, it selects the appropriate channel before
+        checking for the device. If the device is not found, it prints a warning
+        message if show_warnings is True.
+
+        Args:
+            show_warnings (bool, optional): If True, show a warning if the device is not found.
+            Defaults to True.
+        """
+        self.is_mugs_used()
```

Comment:
van een functie die met is_ begint verwacht ik dat hij een boolean returned, maar in dit geval is er een 'side effect'. Kan deze functie een bool returnen zodat je if is_mugs_used() kunt doen?

P.S. mugs -> mux (gebruikelijke afkorting voor multiplexer)